src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useCheckContractArchiveExists/index.spec.tsx

```diff
@@ -0,0 +1,130 @@
+import { faker } from '@faker-js/faker';
+import fetchMock from 'fetch-mock';
+import { act, renderHook, waitFor } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { PropsWithChildren } from 'react';
+import { RichieContextFactory as mockRichieContextFactory } from 'utils/test/factories/richie';
+import JoanieApiProvider from 'contexts/JoanieApiContext';
+import useCheckContractArchiveExists, {
+  CheckContractArchiveStatus,
+  UseCheckContractArchiveExistsProps,
+} from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockRichieContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://auth.test' },
+    joanie_backend: { endpoint: 'https://joanie.test' },
+  }).one(),
+}));
+
+jest.mock('settings', () => ({
+  ...jest.requireActual('settings'),
+  CONTRACT_DOWNLOAD_SETTINGS: {
+    ...jest.requireActual('settings').CONTRACT_DOWNLOAD_SETTINGS,
+    pollInterval: 100,
+  },
+}));
+
+const mockCheckArchive = jest.fn();
+const mockCreateArchive = jest.fn();
+const mockGetArchive = jest.fn();
+jest.mock('hooks/useContractArchive', () => ({
+  __esModule: true,
+  default: () => ({
+    methods: { get: mockGetArchive, create: mockCreateArchive, check: mockCheckArchive },
+  }),
+}));
+
+describe('useCheckContractArchiveExists', () => {
+  const Wrapper = ({ children }: PropsWithChildren) => {
+    return (
+      <IntlProvider locale="en">
+        <JoanieApiProvider>{children}</JoanieApiProvider>
+      </IntlProvider>
+    );
+  };
+  beforeEach(() => {
+    // Joanie providers calls
+    fetchMock.get('https://joanie.test/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.test/api/v1.0/credit-cards/', []);
+    fetchMock.get('https://joanie.test/api/v1.0/addresses/', []);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    fetchMock.restore();
+  });
+
+  it("shouldn't trigger pulling on mount when archive id is given as props", () => {
+    const { result } = renderHook(useCheckContractArchiveExists, {
+      wrapper: Wrapper,
+    });
+    expect(result.current.isContractArchiveExists).toBe(false);
+    expect(result.current.status).toBe(CheckContractArchiveStatus.IDDLE);
+    expect(mockCheckArchive).not.toHaveBeenCalled();
+  });
+
+  it('should trigger pulling on rerender when archive id become set', async () => {
+    const { result, rerender } = renderHook(
+      (props: UseCheckContractArchiveExistsProps) => useCheckContractArchiveExists(props),
+      {
+        wrapper: Wrapper,
+      },
+    );
+
+    mockCheckArchive.mockResolvedValue(false);
+    rerender({ contractArchiveId: faker.string.uuid() });
+
+    await waitFor(() => {
+      expect(result.current.status).toBe(CheckContractArchiveStatus.PULLING);
+    });
+    expect(result.current.isContractArchiveExists).toBe(false);
+    expect(mockCheckArchive).toHaveBeenCalledTimes(1);
+  });
+
+  it('should trigger pulling on mount when archive id is given', async () => {
```

Comment:
Ditto ( the same goes on for any other occurrences, I'm not adding similar comments 👀 )